# HG changeset patch
# User Rik <rik@octave.org>
# Date 1391100582 28800
#      Thu Jan 30 08:49:42 2014 -0800
# Branch gui-release
# Node ID f74c6aaa6d0f968fe113a56236c78e8ae7f32937
# Parent  073fbe7e6a3a75852686026ea0de904182c84a99
Use size_t instead of int in cset 073fbe7e6a3a.

* file-ops.cc (native_separator_path): Return type of length() on C++ string
is size_t, rather than int.

diff --git a/liboctave/system/file-ops.cc b/liboctave/system/file-ops.cc
--- a/liboctave/system/file-ops.cc
+++ b/liboctave/system/file-ops.cc
@@ -358,32 +358,34 @@ file_ops::concat (const std::string& dir
   return dir.empty ()
          ? file
          : (is_dir_sep (dir[dir.length ()-1])
             ? dir + file
             : dir + dir_sep_char () + file);
 }
 
 std::string
-file_ops::native_separator_path(const std::string& path)
+file_ops::native_separator_path (const std::string& path)
 {
   std::string retval;
-  if (dir_sep_char() == '/')
+
+  if (dir_sep_char () == '/')
     retval = path;
   else
     {
-      int n = path.length ();
-      for (int  i = 0; i < n; i++)
+      size_t n = path.length ();
+      for (size_t i = 0; i < n; i++)
         {
-          if (path[i] == '/') 
+          if (path[i] == '/')
             retval += dir_sep_char();
           else
             retval += path[i];
         }
     }
+
   return retval;
 }
 
 int
 octave_mkdir (const std::string& nm, mode_t md)
 {
   std::string msg;
   return octave_mkdir (nm, md, msg);
